src/context.rs

```diff
@@ -1044,4 +1047,48 @@ mod tests {
 
         Ok(())
     }
+
+    #[async_std::test]
+    async fn test_ongoing() -> Result<()> {
+        let context = TestContext::new().await;
+
+        // No ongoing process allocated.
+        assert!(context.shall_stop_ongoing().await);
+
+        // Allocate ongoing process.
+        let receiver = context.alloc_ongoing().await?;
+
+        // Cannot allocate another ongoing process while the first one is running.
+        assert!(context.alloc_ongoing().await.is_err());
+
+        // Stop signal is not sent yet.
+        assert!(receiver.try_recv().is_err());
+
+        // Ongoing process should not be stopped yet.
+        assert!(!context.shall_stop_ongoing().await);
+
+        // Send the stop signal.
+        context.stop_ongoing().await;
+
+        // Receive stop signal.
+        receiver.recv().await?;
+
+        // Ongoing process has to be stopped now.
+        assert!(context.shall_stop_ongoing().await);
```

Comment:
Not sure all of these line comments are needed, some of them (like, these four, "Allocate ongoing process.", "Free ongoing process.") seem to restate what the next line does a bit... OTOH, they don't interfere, so it's fine keeping them, too.